datastore/datastore.go

```diff
@@ -679,3 +701,40 @@ func (c *Client) Mutate(ctx context.Context, muts ...*Mutation) (ret []*Key, err
 	}
 	return ret, nil
 }
+
+// ReadTime specifies a snapshot (time) of the database to read.
+func ReadTime(t time.Time) ReadOption {
+	return docReadTime{t}
+}
+
+type docReadTime struct {
+	time.Time
+}
+
+func (drt docReadTime) apply(rs *readSettings) {
+	rs.readTime = drt.Time
+}
+
+// ReadOption provides specific instructions for how to access documents in the database.
+// Currently, only ReadTime is supported.
+type ReadOption interface {
+	apply(*readSettings)
+}
+
+type readSettings struct {
+	readTime time.Time
+}
+
+// WithReadOptions specifies constraints for accessing documents from the database,
+// e.g. at what time snapshot to read the documents.
+// The client uses this value for subsequent reads, unless additional ReadOptions
+// are provided.
+func (c *Client) WithReadOptions(ro ...ReadOption) *Client {
+	for _, r := range ro {
+		switch o := r.(type) {
```

Comment:
This type switch is not necessary since all your `ro` have the same signature.

Reply:
Done.